pkg/controller/storagecluster/externalstoragecluster_controller.go

```diff
@@ -0,0 +1,317 @@
+			Namespace: sc.Namespace,
+			Labels:    labels,
+		},
+		Spec: cephv1.ClusterSpec{
+			External: cephv1.ExternalSpec{
+				Enable: true,
+			},
+			DataDirHostPath: "/var/lib/rook",
+			CephVersion: cephv1.CephVersionSpec{
+				Image: cephImage,
+			},
+		},
+	}
+	return externalCephCluster
+}
+
+func (r *ReconcileStorageCluster) ensureExternalCephCluster(
+	sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+
+	// Define a new CephCluster object
+	cephCluster := newExternalCephCluster(sc, r.cephImage)
+
+	// Set StorageCluster instance as the owner and controller
+	if err := controllerutil.SetControllerReference(sc, cephCluster, r.scheme); err != nil {
+		return err
+	}
+
+	// Check if this CephCluster already exists
+	found := &cephv1.CephCluster{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cephCluster.Name, Namespace: cephCluster.Namespace}, found)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info("Creating External CephCluster: v", r.cephImage)
+			return r.client.Create(context.TODO(), cephCluster)
+		}
+		return err
+	}
+
+	// Update the CephCluster if it is not in the desired state
+	if !reflect.DeepEqual(cephCluster.Spec, found.Spec) {
+		reqLogger.Info("Updating spec for External CephCluster")
+		sc.Status.Phase = string(found.Status.State)
+		if err := r.client.Status().Update(context.TODO(), sc); err != nil {
+			reqLogger.Error(err, "Failed to add conditions to status")
+			return err
+		}
+		found.Spec = cephCluster.Spec
+		return r.client.Update(context.TODO(), found)
+	}
+
+	// Add it to the list of RelatedObjects if found
+	objectRef, err := reference.GetReference(r.scheme, found)
+	if err != nil {
+		return err
+	}
+	objectreferencesv1.SetObjectReference(&sc.Status.RelatedObjects, *objectRef)
+
+	// Handle CephCluster resource status
+	if found.Status.State == "" {
+		reqLogger.Info("CephCluster resource is not reporting status.")
+		// What does this mean to OCS status? Assuming progress.
+		reason := "CephClusterStatus"
+		message := "CephCluster resource is not reporting status"
+		statusutil.MapCephClusterNoConditions(&r.conditions, reason, message)
+	} else {
+		// Interpret CephCluster status and set any negative conditions
+		// here negative conditions for external cluster has tobe set
+		statusutil.MapExternalCephClusterNegativeConditions(&r.conditions,found)
+	}
+
+	// When phase is expanding, wait for CephCluster state to be updating
+	// this means expansion is in progress and overall system is progressing
+	// else expansion is not yet triggered
+	if sc.Status.Phase == statusutil.PhaseClusterExpanding &&
+		found.Status.State != cephv1.ClusterStateUpdating {
+		r.phase = statusutil.PhaseClusterExpanding
+	}
+
+	if found.Status.State == cephv1.ClusterStateConnecting {
+		sc.Status.Phase = statusutil.PhaseConnecting
+	}
+
+	if found.Status.State == cephv1.ClusterStateConnected {
+		sc.Status.Phase = statusutil.PhaseConnected
+	}
```

Comment:
Is this actually getting reflected on the CR? I don't see it being updated anywhere.

Reply:
It was reflecting in the CR when we tested it. Could you elaborate on the updated part ?

Reply:
These lines are just assigning values to variable. The actual status update is happening somewhere else. I am asking if you are updating it yourself or depending on some other function to update it? (where is the API call to write this to etcd ?)

Reply:
I was thinking that the reconcile logic would be taking care of writing/updating all the corresponding values. If you could point me on how we could do it manually, I'll take of it